add basic api implimentation

EttienneS opened the pull request:
rename old methods
add swashbuckle

EttienneS committed:
add basic api implimentation
rename old methods
add swashbuckle

EttienneS committed:
fix name to use url

EttienneS committed:
Rename method

Stratigix reviewed (commented):
Hey Ett, can you please give a short description of the changes?

EttienneS commented:
Change Summary:

Added two async methods to the ShortenController:

- AddShortUrl that takes in a URL (or any value at this stage) and returns a unique key (a GUID currently)
- GetOriginalUrl that takes a Key as a parameter and returns the original value

These two methods are supported by several Interface changes to create response/request DTO's and to translate between them and the request/response objects.

Finally we added a two methods to the repository layer that actually saves and loads the values to the database.

For this change we renamed most of the methods referencing Kort (Afrikaans for short) to ShortenUrl to make it easier to follow for non-afrikaans speakers.

Stratigix reviewed (approved):
Looks good!
